docs/jetstream.md

```diff
@@ -0,0 +1,214 @@
+        auth:                           # only one of the following keys may be set (see "Future Functionality" below)
+            credentialFile:             # details of the NATS credential file to use for authentication, initially only
+                                        # 'secret' will be supported, but other options may be implemented in the future.
+                secret:
+                    key: ""             # the key of the secret containing the credentials, defaults to "nats.creds"
+                    secretName: ""      # the secret containing a NATS credentials file.
+        tls:
+            secretName: ""              # a secret containing the same keys as the "kubernetes.io/tls" type, if only a 
+                                        # `ca.crt` is present then we skip client verification but still ensure the 
+                                        # connection is encrypted with server verification.
+```
+
+## JetStream integration
+
+This section outlines the features of NATS/JetStream which should be supported by this channel implementation.
+
+### Stream/Consumer configuration
+
+Each `NatsJetStreamChannel` should have the ability to configure the details of the created Streams and Consumers from
+it's `spec` field. For instances where `NatsJetStreamChannel` is configured as the default implementation for `Channel`
+resources, it's up to the user to configure these fields correctly in the `default-ch-webhook` config map.
+
+Channels will by default create a stream name based of the channel namespace/name, however a name can be overridden to
+bind an existing stream which may already contain events.
+
+Consumers will be created with the default configuration:
+
+- Durable - the durable name will be the same as the consumer name. All Knative eventing consumers must be durable to
+  ensure delivery, this cannot be disabled.
+- Pull based - the dispatcher is responsible for pulling messages as it's ready, push-based options cannot be configured
+- Ack Policy is Explicit - this is the only allowed option for pull-based consumers
+
+Other configuration elements use the JetStream defaults, and can be overridden in the CRD.
+
+### Security
+
+The dispatcher will support the following security features:
+
+- Unsecured NATS clusters (i.e. for usage within private networks)
+- Secured NATS clusters with a Credentials File (https://docs.nats.io/developing-with-nats/security/creds)
+    - Globally-scoped credentials by a ConfigMap/Secret in the `knative-eventing` namespace
+    - Namespace-scoped credentials by a ConfigMap/Secret in the same namespace as the `NatsJetStreamChannel`
+    - Channel-scoped credentials specified by a Secret and referenced in the `NatsJetStreamChannel` resource.
+- TLS-enabled clusters (https://docs.nats.io/developing-with-nats/security/tls)
+    - CA should be defined in a ConfigMap/Secret in the `knative-eventing` namespace.
```

Comment:
the `config-nats` configmap should tell the dispatcher the CA, either a base64'd string or a ref to a secret